x-pack/plugins/security/server/authentication/api_keys.ts

```diff
@@ -42,24 +57,103 @@ export interface CreateAPIKeyResult {
   api_key: string;
 }
 
-export async function createAPIKey({
-  body,
-  loggers,
-  callAsCurrentUser,
-  isSecurityFeatureDisabled,
-}: CreateAPIKeyOptions): Promise<CreateAPIKeyResult | null> {
-  const logger = loggers.get('api-keys');
+/**
+ * The return value when invalidating an API key in Elasticsearch.
+ */
+export interface InvalidateAPIKeyResult {
+  /**
+   * The IDs of the API keys that were invalidated as part of the request.
+   */
+  invalidated_api_keys: string[];
+  /**
+   * The IDs of the API keys that were already invalidated.
+   */
+  previously_invalidated_api_keys: string[];
+  /**
+   * The number of errors that were encountered when invalidating the API keys.
+   */
+  error_count: number;
+  /**
+   * Details about these errors. This field is not present in the response when error_count is 0.
+   */
+  error_details?: Array<{
+    type: string;
+    reason: string;
+    caused_by: {
+      type: string;
+      reason: string;
+    };
+  }>;
+}
+
+/**
+ * Class responsible for managing Elasticsearch API keys.
+ */
+export class APIKeys {
+  private readonly logger: Logger;
+  private readonly clusterClient: PublicMethodsOf<ClusterClient>;
+  private readonly isSecurityFeatureDisabled: () => boolean;
+
+  constructor({ logger, clusterClient, isSecurityFeatureDisabled }: ConstructorOptions) {
+    this.logger = logger;
+    this.clusterClient = clusterClient;
+    this.isSecurityFeatureDisabled = isSecurityFeatureDisabled;
+  }
 
-  if (isSecurityFeatureDisabled()) {
-    return null;
+  /**
+   * Tries to create an API key for the current user.
+   * @param param0 The options to create an API key
+   */
+  async create(
+    request: KibanaRequest,
+    body: CreateAPIKeyParams
+  ): Promise<CreateAPIKeyResult | null> {
+    if (this.isSecurityFeatureDisabled()) {
+      return null;
+    }
+
+    this.logger.debug('Trying to create an API key');
+
+    // User needs `manage_api_key` privilege to use this API
+    const key = (await this.clusterClient
```

Comment:
__optional nit:__ we can also wrap this into `try/catch` like we have in `invalidate`, just for the consistency sake. It's out of the scope of this PR, hence up to you.